### PR TITLE
Use real name and email in cherrypicker

### PR DESCRIPTION
**Pull request by kargakis**
/cc stevekuznetsov 

We will need to make the bot email public in its profile for this to work.

### CODE REVIEW

**Comment by kargakis** [comment]
Example run: https://github.com/kargakis/grin/pull/7

File: prow/github/client.go [hunk 1]
```diff
@@ -391,10 +392,36 @@ func (c *Client) BotName() (string, error) {
...
+		// https://developer.github.com/v3/users/#get-a-single-user
+		c.email = u.Email
 	}
 	return c.botName, nil
 }
 
+func (c *Client) Email() (string, error) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if c.email == "" {
```
Comment (stevekuznetsov) [inline prow/github/client.go]: can we shove this `if` block into a shared method? seems 100% the same between these two?

Comment (kargakis) [reply prow/github/client.go]: sure, fixed.

**Pull request merged**